crates/bevy_pbr/src/light.rs

```diff
@@ -480,14 +481,61 @@ fn cluster_to_index(cluster_dimensions: UVec3, cluster: UVec3) -> usize {
     ((cluster.y * cluster_dimensions.x + cluster.x) * cluster_dimensions.z + cluster.z) as usize
 }
 
+// Sort point lights with shadows enabled first, then by a stable key so that the index
+// can be used to render at most `MAX_POINT_LIGHT_SHADOW_MAPS` point light shadows, and
+// we keep a stable set of lights visible
+pub(crate) fn point_light_order(
+    (entity_1, shadows_enabled_1): (&Entity, &bool),
+    (entity_2, shadows_enabled_2): (&Entity, &bool),
+) -> std::cmp::Ordering {
+    shadows_enabled_1
+        .cmp(shadows_enabled_2)
+        .reverse()
+        .then_with(|| entity_1.cmp(entity_2))
+}
+
 // NOTE: Run this before update_point_light_frusta!
 pub fn assign_lights_to_clusters(
     mut commands: Commands,
     mut global_lights: ResMut<VisiblePointLights>,
     mut views: Query<(Entity, &GlobalTransform, &Camera, &Frustum, &mut Clusters)>,
     lights: Query<(Entity, &GlobalTransform, &PointLight)>,
 ) {
-    let light_count = lights.iter().count();
+    let mut lights = lights.iter().collect::<Vec<_>>();
+
+    if lights.len() > MAX_POINT_LIGHTS {
+        lights.sort_by(|(entity_1, _, light_1), (entity_2, _, light_2)| {
+            point_light_order(
+                (entity_1, &light_1.shadows_enabled),
+                (entity_2, &light_2.shadows_enabled),
+            )
+        });
+
+        // check each light against each view's frustum, keep only those that affect at least one of our views
+        let frusta: Vec<_> = views.iter().map(|(_, _, _, frustum, _)| *frustum).collect();
+        lights = lights
+            .into_iter()
+            .filter(|(_, light_transform, light)| {
+                let light_sphere = Sphere {
+                    center: light_transform.translation,
+                    radius: light.range,
+                };
+
+                frusta
+                    .iter()
+                    .any(|frustum| frustum.intersects_sphere(&light_sphere))
+            })
+            // take one extra light to check if we should emit the warning
+            .take(MAX_POINT_LIGHTS + 1)
+            .collect();
+
+        if lights.len() > MAX_POINT_LIGHTS {
+            warn!("MAX_POINT_LIGHTS ({}) exceeded", MAX_POINT_LIGHTS);
```

Comment:
not a fan of logs that will potentially happens on every frame

Reply:
i agree, but it should definitely be emitted once. 

is it reasonable to add a Local<bool> to check if the warning has been emitted already?

Reply:
The problem is that if there are other logs then it may get lost. I think we need a general pattern for this. Perhaps something like a Local<> that holds the last time we logged, and then just log once every 10s or something?

Reply:
no, I would keep it that way for now rather than add an additional parameter... there are a few other places with similar issues, we don't have the good solution for that yet

Reply:
ideally the logger should be able to detect duplicate logs and only log them once in a while 🤔

Reply:
For something that should be "fixed" i'm cool with warning every frame (until we can suppress more intelligently). But in this case, complex scenes could easily go over the max point light limit. This isn't "wrong" and I dont think we should be aggressively pushing people to hide lights that go over the limit.

Manually suppressing duplicate warnings is important to do here imo.

Reply:
@cart what do you think of my suggestion above as an approach to 'rate limiting' the log?

Reply:
limited to warning once for now

Reply:
@superdump  yeah i like that, although I'm cool with logging once for now. "rate limiting" is an option, but we could also consider detecting fluctuations in light counts (ex: if you go under the limit and then back over, we print the warning again).